exercises/robot-name/robot-name-test.rkt

```diff
@@ -0,0 +1,38 @@
+#lang racket
+(require racket/class
```

Comment:
I should learn how to use classes in Racket, but I wonder if it isn't better to leave it open ended.  It would probably mean requiring the solution provide some wrapper functions that let the student use whatever backend implementation they want.

The wider Racket community might disagree with me, but since it is a Scheme I think it would be really nice if it were also made possible for me (as a student) to use records or even to hand roll my robot objects the old fashion way with state and message passing.

Reply:
The types would be like:
```
new-robot -> robot
robot-name : robot -> string
robot-reset  : robot ; just for side-effects
reset-name-pool ; just for side-effects
```
Sorry for the ML style types, I started writing them like `typed/racket` and realiszed that I didn't understand it enough

Reply:
Anyway, let me know what you think about this suggestion.

Reply:
I have always loved how flexible Lisp-y languages are, and I was a bit worried about constraining the problem to Racket's classes (even though they are pretty cool). I was mostly trying to make life easy for me when writing the test module.

I don't (yet) program in ML, but I appreciate what you are trying to say. In terms of specifying the "contract" for the program, I do like your suggestion - and the generality of answer that it allows the student to submit I like.